### DISCUSSION TITLE
[style-loader] when debugging, why the code DID NOT jump into file injectStylesIntoStyleTag.js ?

### DISCUSSION THREAD

**Discussion by wine-fall** [webpack/webpack#16822] [Q&A] [answered]
Hi, guys:

I'm debugging the `style-loader` source code, here are what I did:

1. add a break point in file: 'src/runtime/injectStylesIntoStyleTag.js'
![image](https://user-images.githubusercontent.com/62830944/224949999-ac5e8a3c-3365-41e1-a140-95ec3ac30914.png)

2. go to the file: 'test/base-option.test.js' and click 'debug'

![image](https://user-images.githubusercontent.com/62830944/224950196-420f331f-7bfe-44d8-933e-253ee700665c.png)

And the code didn't break in `injectStylesIntoStyleTag.js`, why???

why I think it will be?

1. I read the code in file 'src/index.js' and on line 161, It import a function called `API`, right?
2. And on line 189, call the function `API` and return a function called `update`

![image](https://user-images.githubusercontent.com/62830944/224950728-085f57eb-8c06-4864-b71e-82053caa74b6.png)

**Comment by alexander-akait** [comment, marked as answer]
Hello, yes, can you descibe your problem, why you need to debug it?

**Reply by wine-fall** [reply]
> Hello, yes, can you descibe your problem, why you need to debug it?

HI @alexander-akait : 

Thanks for replying.

I'm just learning how to write a loader and how does a loader work, so I want to debug it. (not for any bugs) 

By the way, I see `style-loader` translate `.css` to a string looks like a `.js`, will webpack run this `.js` like string?

**Reply by alexander-akait** [reply]
style-loader is not simple things where I recommend to start :smile: But yes `css-loader` transilte CSS into JS pseudo modules https://github.com/webpack-contrib/css-loader/blob/master/src/runtime/api.js#L6, then `style-loader` apply them on your document (it is based on your options), but code is not easy to read due mix of runtime and internal code

**Reply by wine-fall** [reply]
hi @alexander-akait :

Thank you and as you suggest I'm going to find another loader to learn. Do you have any advices? (both plugin and loader 😭) 🙏

**Reply by alexander-akait** [reply]
You can look at `css-loader`, if you want to look at CSS pipeline, it is easier to undestand
